src/lib.rs

```diff
@@ -72,3 +80,37 @@ fn enumerate_project_files(language: &dyn SupportedLanguage) -> Vec<DirEntry> {
         .filter(|entry| entry.metadata().unwrap().is_file())
         .collect()
 }
+#[derive(Debug)]
+struct TreeSitterMatcher {
+    matches: Mutex<Vec<Match>>,
+}
+
+impl TreeSitterMatcher {
+    fn new(mut matches: Vec<Match>) -> Self {
+        matches.sort_by_key(|m| m.start());
+
+        Self {
+            matches: Mutex::new(matches),
+        }
+    }
+}
+
+impl Matcher for TreeSitterMatcher {
+    type Captures = NoCaptures;
+
+    type Error = NoError;
+
+    fn find_at(&self, _haystack: &[u8], at: usize) -> Result<Option<Match>, Self::Error> {
+        let _match = self
+            .matches
+            .lock()
+            .unwrap()
+            .pop()
+            .map(|m| Match::new(m.start() - at, m.end() - at));
+        Ok(_match)
```

Comment:
The function params are:
- `haystack`: remaining bytes to search
- `at`: offset into the file where `haystack` starts, I think

It wants matches returned relative to the start of the haystack, so subtrack `at` from the "real" matches. (`at` has always been 0 in my experiments, though)